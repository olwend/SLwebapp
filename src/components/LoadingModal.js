import React from 'react'

import { Modal, ModalHeader, ModalBody, Progress } from 'reactstrap'

const LoadingModal = (props) => (
  <Modal isOpen={props.dpgLoading}>
    <ModalHeader>Labs Setup</ModalHeader>
    <ModalBody className="mb-3">
      <div className="progress-container">
        <Progress max="100" value={props.pbValue}>
          <span className="progress-value" style={{fontSize: "10px"}}>{props.pbValue}%</span>
        </Progress>
      </div>
    </ModalBody>
  </Modal>
);


export default LoadingModal
