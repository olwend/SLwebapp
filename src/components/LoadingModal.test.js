import React from 'react'

import { shallow } from 'enzyme'

import LoadingModal from './LoadingModal'

test('Renders properly', () => expect(shallow(<LoadingModal />)).toMatchSnapshot())
