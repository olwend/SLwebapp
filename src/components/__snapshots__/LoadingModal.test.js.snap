// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  returnFocusAfterClose={true}
  role="dialog"
  scrollable={false}
  unmountOnClose={true}
  zIndex={1050}
>
  <ModalHeader
    charCode={215}
    closeAriaLabel="Close"
    tag="h5"
    wrapTag="div"
  >
    Labs Setup
  </ModalHeader>
  <ModalBody
    className="mb-3"
    tag="div"
  >
    <div
      className="progress-container"
    >
      <Progress
        max="100"
        tag="div"
        value={0}
      >
        <span
          className="progress-value"
          style={
            Object {
              "fontSize": "10px",
            }
          }
        >
          %
        </span>
      </Progress>
    </div>
  </ModalBody>
</Modal>
`;
